rules/REQUEST-944-APPLICATION-ATTACK-JAVA.conf

```diff
@@ -0,0 +1,290 @@
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML|XML:/* \
+    "@rx (?:\xac\xed\x00\x05)" \
+    "id:944300,\
+    phase:2,\
+    pass,\
+    msg:'Magic bytes Detected, probable java serialization in use',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    severity:'NOTICE',\
+    setvar:'tx.msg=%{rule.msg}',\
+    setvar:tx.rce_score=+%{tx.notice_anomaly_score},\
+    setvar:tx.anomaly_score=+%{tx.notice_anomaly_score},\
+    setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{tx.0}'"
+
+# Detecting possibe base64 text to match encoded magic bytes \xac\xed\x00\x05 with padding encoded in base64 strings are rO0ABQ KztAAU Cs7QAF
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML|XML:/* \
+    "@rx (?:rO0ABQ|KztAAU|Cs7QAF)" \
+    "id:944310,\
+    phase:2,\
+    pass,\
+    msg:'Magic bytes Detected Base64 Encoded, probable java serialization in use',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    severity:'NOTICE',\
+    setvar:'tx.msg=%{rule.msg}',\
+    setvar:tx.rce_score=+%{tx.notice_anomaly_score},\
+    setvar:tx.anomaly_score=+%{tx.notice_anomaly_score},\
+    setvar:'tx.%{rule.id}-OWASP_CRS/WEB_ATTACK/RCE-%{matched_var_name}=%{tx.0}'"
+
+# Looking for possibly RCE vulnerable java classes,
+# anomaly score set to notice as both conditions may indicate a possibly vulnerable application
+SecRule ARGS|ARGS_NAMES|REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_BODY|REQUEST_HEADERS|XML|XML:/* \
+    "@rx (?:[a-zA-Z0-9\-_]{45})(?:[a-zA-Z0-9\-_]{3})*(?:[a-zA-Z0-9\-_]{1}==|[a-zA-Z0-9\-_]{2}=)?" \
+    "id:944320,\
+    phase:2,\
+    block,\
+    t:base64Decode,t:lowercase,\
+    msg:'Probable vulnerable java class in use',\
+    tag:'application-multi',\
+    tag:'language-java',\
+    tag:'platform-multi',\
+    tag:'attack-rce',\
+    tag:'OWASP_CRS/WEB_ATTACK/COMMAND_INJECTION',\
+    tag:'WASCTC/WASC-31',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    rev:'1',\
+    ver:'OWASP_CRS/3.1.0',\
+    severity:'NOTICE',\
+    chain"
+        SecRule MATCHED_VARS "@rx (?:runtime|processbuilder|clonetransformer|forclosure|instantiatefactory|instantiatetransformer|invokertransformer|prototypeclonefactory|prototypeserializationfactory|whileclosure)" \
```

Comment:
This 'SecRule' should be aligned with the previous 'chain'

Reply:
fixed, missed that one